test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingFromFormTest.cs

```diff
@@ -0,0 +1,170 @@
+                new KeyValuePair<string, string>("customPrefix.Employees[0].EmployeeSSN", "123132131"),
+            };
+            request.Content = new FormUrlEncodedContent(nameValueCollection);
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            var body = await response.Content.ReadAsStringAsync();
+            var company = JsonConvert.DeserializeObject<Company>(body);
+
+            var employee = Assert.Single(company.Employees);
+            Assert.Equal("123132131", employee.SSN);
+        }
+
+        [Fact]
+        public async Task FromFormAttribute_CustomModelPrefix_ForCollectionProperty()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+
+            // [FromForm(Name = "TestEmployees")] is used to apply a prefix
+            var url =
+                "http://localhost/FromFormAttribute_Company/CreateDepartment";
+            var request = new HttpRequestMessage(HttpMethod.Post, url);
+            var nameValueCollection = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("department.TestEmployees[0].EmployeeSSN", "123132131"),
+            };
+            request.Content = new FormUrlEncodedContent(nameValueCollection);
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            var body = await response.Content.ReadAsStringAsync();
+            var department = JsonConvert.DeserializeObject<
+                FromFormAttribute_CompanyController.FromForm_Department>(body);
+
+            var employee = Assert.Single(department.Employees);
+            Assert.Equal("123132131", employee.SSN);
+        }
+
+        [Fact]
+        public async Task FromForm_NonExistingValueAddsValidationErrors_OnProperty_UsingCustomModelPrefix()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+
+            // [FromForm(Name = "TestEmployees")] is used to apply a prefix
+            var url =
+                "http://localhost/FromFormAttribute_Company/ValidateDepartment";
+            var request = new HttpRequestMessage(HttpMethod.Post, url);
+
+            // No values.
+            var nameValueCollection = new List<KeyValuePair<string, string>>();
+            request.Content = new FormUrlEncodedContent(nameValueCollection);
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            var body = await response.Content.ReadAsStringAsync();
+            var result = JsonConvert.DeserializeObject<Result>(body);
+            Assert.Null(result.Value);
+            var error = Assert.Single(result.ModelStateErrors);
+            Assert.Equal("TestEmployees", error);
+        }
+
+        private class Result
```

Comment:
it was a nit but FYI our usual max. count of duplicates before we share is around 2 or 3.